### PR TITLE
docs: Added Markdown files for Documentation

### LINKED ISSUE #273: [FEATURE] Add docs for the Website
### Is your feature request related to a problem? Please describe.

The README file currently lacks examples and use cases for both Drifty CLI and GUI. It also does not mention the custom issue labels that the project uses. 
Before testing their changes, the contributors need to run some commands. The addition of Dockerfiles to this project via #218 will require some notes and advices, but the README file is already too crowded and lengthy.

### Describe the solution you'd like.

I would like to have a docs section for Drifty with sections like - **Quickstart**, **Getting Started**, **Contributing**, **FAQ** and **Troubleshooting**.
- **Quickstart**: A brief introduction to the main features of Drifty, with a simple example to get users up and running in minutes.
- **Getting Started**: A more detailed guide on how to install, configure, and use Drifty, with explanations of the core components and options.
- **Contributing**: A section for developers who want to help make Drifty better, with instructions on how to set up a development environment, submit pull requests, and report bugs and new ideas.
- **FAQ**: A collection of common questions and answers about Drifty, covering topics such as compatibility, performance, security, and customization.
- **Troubleshooting**: This subsection helps to resolve common issues and errors that users may encounter while using Drifty and suggests possible solutions and workarounds.

The docs should be in **markdown files** and will be rendered by next js. You can read more about this [here](https://nextjs.org/docs/app/building-your-application/configuring/mdx).
> If you have any suggestions to add more suggestions, please let me know.

### Screenshots

The docs page can have a look like this -

Image

> You can see the live version of the docs homepage [here](https://drifty-git-preview-docs-saptarshisarkar12.vercel.app/docs).

### Additional information
*Skills you need to solve this issue* - **NextJS**, **TailwindCSS**, [**MD / MDX in Nextjs**](https://nextjs.org/docs/app/building-your-application/configuring/mdx).
Those who wish to work on this issue, please leave a comment below and I will assign you with the task.

### PR DESCRIPTION
**Pull request by hasnainmakada-99**
Fixes #273 

## Changes proposed

Hey, @SaptarshiSarkar12 I've currently added FAQ, getting started and Quickstart guide, I will add more docs related to the aspects mentioned in issue #273, I am currently raising this PR just to break the ICE of this Stale ISSUE, Hope you'll understand it. More changes will be pushed and reviewing will be needed

## Check List (Check all the applicable boxes)

- [x] My code follows the code style of this project.
- [x] My change requires changes to the documentation.
- [x] I have updated the documentation accordingly.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Screenshots
NaN

## Note to reviewers
NaN

## Summary by CodeRabbit

- **New Features**
	- Added a Frequently Asked Questions (FAQ) section to enhance user support for the Drifty project.
	- Introduced a Quickstart guide for easy downloading and installation of the Drifty application.
	- Launched a comprehensive guide for community contributions, including a Code of Conduct and issue reporting process.
	- Released detailed documentation on the project's architecture and building executable binaries.
	- Expanded guidance on running the Drifty GUI Docker image on macOS.
	- Enhanced the overview of the Drifty GUI with detailed descriptions of its components and functionalities.

- **Documentation**
	- Updated the Getting Started guide with installation instructions for Drifty GUI and CLI.
	- Provided extensive documentation for Drifty CLI and GUI, detailing downloading processes and interface navigation.

### CODE REVIEW

**Comment by SaptarshiSarkar12** [comment]
@hasnainmakada-99 Please add one blank line after **## Main Features of Drifty** (line 7 of `Quickstart.md` file) and **##  A simple demo on how to get started with the GUI of Drifty** (line 16 of `Quickstart.md` file) to fix the linter errors.

**Comment by SaptarshiSarkar12** [comment]
@hasnainmakada-99 I am sorry. I mistakenly wrote the heading (**A simple demo on how to get started with the GUI of Drifty**) as line 16 of `Quickstart.md` file. It is present in line number 17 of that file. Please add a line break after any heading.

**Comment by SaptarshiSarkar12** [comment]
@hasnainmakada-99 What is the progress of this PR?

**Comment by hasnainmakada-99** [comment]
Hey @SaptarshiSarkar12, Sorry for the late reply actually I was gone out of town as my exams just got over, So I would like to ask you that so far we I have done the FAQ, Getting Started and Quick Start, so do we need any further guides to make for this project

**Comment by SaptarshiSarkar12** [comment]
It's okay @hasnainmakada-99. No problem :smile:.
We need to add **Contributing** and **Troubleshooting** pages. I can help you with the Contributing page which will describe what is the use of each labels in issues, use of GitHub Projects as our Roadmap, etc. and Troubleshooting which will include common problems faced while installing or using the application.
Some more pages can be added by me later on - I'll think on this once you make everything ready from your end.
@hasnainmakada-99 Let me know if anything else you think can be added :smiley:.

**Comment by hasnainmakada-99** [comment]
Hey @SaptarshiSarkar12 , We can start with the contributing doc right now WDYS? once the contributing docs will be completed then we'll start with the troubleshooting docs

**Comment by SaptarshiSarkar12** [comment]
Yeah, we can start with that. Agreed @hasnainmakada-99 :+1:.

**Comment by hasnainmakada-99** [comment]
Alright, Then tell me whatt should I include inside the contribution guide @SaptarshiSarkar12

**Comment by SaptarshiSarkar12** [comment]
@hasnainmakada-99 The **Contributing** page should contain :point_down: 
- A link to the [Contributing Guidelines](https://github.com/SaptarshiSarkar12/Drifty/blob/master/CONTRIBUTING.md).
- Project structure (like which directory contains which type of files) [You can find an example [here](https://github.com/oracle/graal?tab=readme-ov-file#repository-structure)]
- A brief description of what each issue and PR labels mean (You can keep the heading as **Labels** and sub-headings as 
**In Issues** and **In Pull Requests**)
- A guide on how to Build Drifty executable using two ways (**Local Development** and **Using Docker Compose**)

@hasnainmakada-99 Anything else to add? Please let me know. I'm out of ideas now :laughing:

**Comment by hasnainmakada-99** [comment]
> @hasnainmakada-99 The **Contributing** page should contain 👇
> 
> * A link to the [Contributing Guidelines](https://github.com/SaptarshiSarkar12/Drifty/blob/master/CONTRIBUTING.md).
> * Project structure (like which directory contains which type of files) [You can find an example [here](https://github.com/oracle/graal?tab=readme-ov-file#repository-structure)]
> * A brief description of what each issue and PR labels mean (You can keep the heading as **Labels** and sub-headings as
>   **In Issues** and **In Pull Requests**)
> * A guide on how to Build Drifty executable using two ways (**Local Development** and **Using Docker Compose**)
> 
> @hasnainmakada-99 Anything else to add? Please let me know. I'm out of ideas now 😆

No worries @SaptarshiSarkar12, I'll start working on it ASAP

**Comment by SaptarshiSarkar12** [comment]
Okay @hasnainmakada-99 :smile:.

**Comment by hasnainmakada-99** [comment]
Hey @SaptarshiSarkar12, I'm Halfway done with the contribution docs

**Comment by SaptarshiSarkar12** [comment]
Okay @hasnainmakada-99. That's great :+1:.
Please consider updating your code with the master branch of the original repo :upside_down_face:.

**Comment by hasnainmakada-99** [comment]
Done @SaptarshiSarkar12

**Comment by SaptarshiSarkar12** [comment]
Great :+1: @hasnainmakada-99

**Comment by SaptarshiSarkar12** [comment]
@hasnainmakada-99 Now, would you start with the **troubleshooting** docs page?

**Comment by SaptarshiSarkar12** [comment]
@hasnainmakada-99 Can you please create the feature to render markdown docs pages on the website, before making the Troubleshooting docs? I want to see how it would look like :smile:.

**Comment by hasnainmakada-99** [comment]
> @hasnainmakada-99 Can you please create the feature to render markdown docs pages on the website, before making the Troubleshooting docs? I want to see how it would look like 😄.

Ahh, Actually I am not familiar with implementing that feature @SaptarshiSarkar12, So I may not be able to help you with that

**Comment by SaptarshiSarkar12** [comment]
I guess you haven't understood my words. I was asking you to do the rest of the work of this issue, other than creating md files.

**Comment by hasnainmakada-99** [comment]
> I guess you haven't understood my words. I was asking you to do the rest of the work of this issue, other than creating md files.

Yes @SaptarshiSarkar12, I don't know how to do that, so I can't help you with that, but for now I can work on creating the MD files required to render

**Comment by SaptarshiSarkar12** [comment]
Okay. You can do that. But, the issue was supposed to make the rendering of those files. Still, are you going to learn that?
I have already prepared a protoype of the docs https://github.com/SaptarshiSarkar12/Drifty/tree/preview-docs

**Comment by hasnainmakada-99** [comment]
No @SaptarshiSarkar12, I might not be a good fit for designing that, honestly saying because it will be a waste of time for us, So you should look for someone else to work on the rendering part and I can help you with the docs

**Comment by SaptarshiSarkar12** [comment]
Okay @hasnainmakada-99

**Comment by SaptarshiSarkar12** [comment]
@hasnainmakada-99 Then, let's start with the **Troubleshooting** docs.

**Comment by hasnainmakada-99** [comment]
@SaptarshiSarkar12 So sorry for not updating, actually vivas are going that's why, hope you'll understand

**Comment by hasnainmakada-99** [comment]
Currently the contribution docs are still in the queue, So i'll complete it by today ASAP @SaptarshiSarkar12

**Comment by SaptarshiSarkar12** [comment]
> @SaptarshiSarkar12 So sorry for not updating, actually vivas are going that's why, hope you'll understand

It's fine @hasnainmakada-99. No problem :smile:.

> Currently the contribution docs are still in the queue, So i'll complete it by today ASAP @SaptarshiSarkar12

That's great :+1:. Also, if you have any suggestion on more sections in the docs, please let me know. Suggestions are warmly welcomed :grin:.

**Comment by SaptarshiSarkar12** [comment]
@hasnainmakada-99 How much work is done? Any issues you faced? Please let me know.

**Comment by hasnainmakada-99** [comment]
Hey @SaptarshiSarkar12 , I'll be done with the troubleshooting as well as the quick start docs by monday

**Comment by SaptarshiSarkar12** [comment]
Okay. That's great :+1:. Thank you @hasnainmakada-99!

**Comment by hasnainmakada-99** [comment]
@SaptarshiSarkar12 Will you please check the contribution as well as the quick start guide

At first I created the troubleshooting guide but then I looked at the FAQ guide and it was almost covering all the errors and help needed for the user, So i thought of not to proceed as it will unnecessarily increase the number of documents the page has to load, What are your views on this ? @SaptarshiSarkar12

**Comment by SaptarshiSarkar12** [comment]
Sorry, I am out of station. So, I won't get time upt Thursday. Will check as soon as I return.

**Comment by hasnainmakada-99** [comment]
No worries @SaptarshiSarkar12 😃

File: Website/app/docs/Quickstart.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# A general overview of Dirfty
+
+Dirfty is an open source Interactive file downloader software built using Java and NextJS, Its backend is powered by Java and the front end consists of NextJS and TailwindCSS
+
+With the help of Dirfty, you can download files and videos from various sources such as YouTube, Instagram etc.. You can also download files by providing the link address if it's downloading way too slowly inside your browser.
```
Comment (SaptarshiSarkar12) [inline Website/app/docs/Quickstart.md:5]:
```suggestion
Drifty is an Open-Source Interactive File Downloader software built using Java. The Website is built using NextJS and Tailwind CSS.

With the help of Drifty, you can download files and videos from various sources such as YouTube, Instagram etc.
```

File: Website/app/docs/Quickstart.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+## Main Features of Drifty
+
+- Free and Open Source forever
+- Faster downloading of files, just like IDM
```
Comment (SaptarshiSarkar12) [inline Website/app/docs/Quickstart.md:10]:
```suggestion
- Faster downloading of files
```

File: Website/app/docs/Quickstart.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+Dirfty is an open source Interactive file downloader software built using Java and NextJS, Its backend is powered by Java and the front end consists of NextJS and TailwindCSS
+
+With the help of Dirfty, you can download files and videos from various sources such as YouTube, Instagram etc.. You can also download files by providing the link address if it's downloading way too slowly inside your browser.
+
+## Main Features of Drifty
+
+- Free and Open Source forever
+- Faster downloading of files, just like IDM 
+- Youtube, Instagram, Twitter/X etc.. Platforms supported
```
Comment (SaptarshiSarkar12) [inline Website/app/docs/Quickstart.md:11]: Twitter is not yet supported but we'll think about it in the future :smile:.
```suggestion
- Popular platforms like YouTube, Instagram, etc. are also supported
```

File: Website/app/docs/Quickstart.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+- Both CLI (Command Line Interface) and GUI (Graphical User Interface) are available
+- Parallel downloading support provided
```
Comment (SaptarshiSarkar12) [inline Website/app/docs/Quickstart.md:13]:
```suggestion
- Multi-threaded downloading is used in Drifty to get accelerated download speeds
```

File: Website/app/docs/Quickstart.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+# Let's get you running
+
+##  A simple demo on how to get started with the GUI of Drifty
```
Comment (SaptarshiSarkar12) [inline Website/app/docs/Quickstart.md:17]: The Drifty CLI demo is not covered. Please consider adding it :grin:.

File: Website/app/docs/Quickstart.md [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+https://github.com/hasnainmakada-99/Drifty/assets/82728823/980cafd3-6e3e-4b56-a890-b96ec47b7593
```
Comment (SaptarshiSarkar12) [inline Website/app/docs/Quickstart.md:19]: @hasnainmakada-99 I think you can remove the sound and add texts in the video to show what you are pointing to in the video. You can remove redundant stops like the user is waiting on the homepage of the Drifty Website for more than 25 seconds. Generally, the audio is muted in videos. So, it's better to include your words in the form of texts in the video. 
Two benefits are there :point_down: 
- Improving accessibility
- Reducing size of the video a bit

**Review by SaptarshiSarkar12 — CHANGES_REQUESTED** [review]
@hasnainmakada-99 Please make the requested changes and let me know your ideas if you have any.

File: Website/app/docs/Quickstart.md (hunk 6 above)
Comment (hasnainmakada-99) [reply Website/app/docs/Quickstart.md:19]: Yeah that's true @SaptarshiSarkar12 what you are saying, But I am really bad with editing videos, so can't help you out on this one

File: Website/app/docs/Quickstart.md (hunk 5 above)
Comment (hasnainmakada-99) [reply Website/app/docs/Quickstart.md:17]: But the CLI will be only covered for the windows machine @SaptarshiSarkar12 as I am running on windows that's why

File: Website/app/docs/Quickstart.md (hunk 6 above)
Comment (SaptarshiSarkar12) [reply Website/app/docs/Quickstart.md:19]: Okay. Will try to do that myself if possible. I'm also not too good at editing :upside_down_face:.

File: Website/app/docs/Quickstart.md (hunk 5 above)
Comment (SaptarshiSarkar12) [reply Website/app/docs/Quickstart.md:17]: I'll provide you some resources and texts. Then, can you do it :grin:?

**Comment by SaptarshiSarkar12** [comment]
@hasnainmakada-99 Please make the requested changes 😃.

File: Website/app/docs/Quickstart.md (hunk 6 above)
Comment (SaptarshiSarkar12) [reply Website/app/docs/Quickstart.md]: Done :ballot_box_with_check:!
@hasnainmakada-99 You can see it below. Suggestions are welcomed :smiley:.

https://github.com/SaptarshiSarkar12/Drifty/assets/105960032/5bb21bf6-fb2b-4374-8d0b-14cac3c06908

**Comment by hasnainmakada-99** [comment]
Sure @SaptarshiSarkar12

File: Website/app/docs/Quickstart.md (hunk 6 above)
Comment (hasnainmakada-99) [reply Website/app/docs/Quickstart.md]: Looks great @SaptarshiSarkar12 , I think that we should include this inside the main deployment

Comment (SaptarshiSarkar12) [reply Website/app/docs/Quickstart.md]: Yeah, can you please include that change in your next push?

**Comment by SaptarshiSarkar12** [comment]
@hasnainmakada-99 Please make the requested changes :smile:.

**Comment by hasnainmakada-99** [comment]
> @hasnainmakada-99 Please make the requested changes 😄.

Sure @SaptarshiSarkar currently having a lot of academic pressure so cant cope up with the timings

**Comment by SaptarshiSarkar12** [comment]
@hasnainmakada-99 Okay. Take your time :smiley:.

**Comment by SaptarshiSarkar12** [comment]
@hasnainmakada-99 Are you free this week? Will you be able to complete the task 😁?

**Comment by SaptarshiSarkar12** [comment]
> @hasnainmakada-99 Are you free this week? Will you be able to complete the task 😁?

@hasnainmakada-99 Please reply.

**Comment by hasnainmakada-99** [comment]
Hey @SaptarshiSarkar12 so sorry for late replying, as I am so much caught up with my exams I will not be solve this issue, so sorry once again

**Comment by SaptarshiSarkar12** [comment]
@hasnainmakada-99 It's fine :smile:. So, can I complete the few changes left to be done? And once done, can I merge it?

**Comment by SaptarshiSarkar12** [comment]
@hasnainmakada-99 Thank you :smile:.

**Review by SaptarshiSarkar12 — APPROVED** [review]
@hasnainmakada-99 **Looks good to merge 👍.** 
**Thanks for contributing 🚀 🚀.**
You may join our _Discord server_ - **https://discord.gg/DeT4jXPfkG** to get updates about the project.

**Pull request merged**